ambry-account/src/test/java/com/github/ambry/account/mysql/AccountDaoTest.java

```diff
@@ -0,0 +1,95 @@
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Properties;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+
+/** Unit test for AccountDao class */
+@RunWith(MockitoJUnitRunner.class)
+public class AccountDaoTest {
+
+  private final short accountId = 101;
+  private final String accountName = "samza";
+  private final Account testAccount;
+  private final String accountJson;
+  private final MySqlDataAccessor dataAccessor;
+  private final Connection mockConnection;
+  private final AccountDao accountDao;
+
+  public AccountDaoTest() throws SQLException {
+    testAccount = new AccountBuilder(accountId, accountName, Account.AccountStatus.ACTIVE).build();
+    accountJson = AccountSerdeUtils.accountToJson(testAccount, true);
+    mockConnection = mock(Connection.class);
+    PreparedStatement mockInsertStatement = mock(PreparedStatement.class);
+    when(mockConnection.prepareStatement(contains("insert into"))).thenReturn(mockInsertStatement);
+    when(mockInsertStatement.executeUpdate()).thenReturn(1);
+    PreparedStatement mockQueryStatement = mock(PreparedStatement.class);
+    when(mockConnection.prepareStatement(startsWith("select"))).thenReturn(mockQueryStatement);
+    ResultSet mockResultSet = mock(ResultSet.class);
+    when(mockResultSet.next()).thenReturn(true).thenReturn(false);
+    when(mockResultSet.getString(eq(AccountDao.ACCOUNT_INFO))).thenReturn(accountJson);
+    when(mockQueryStatement.executeQuery()).thenReturn(mockResultSet);
+    dataAccessor = getDataAccessor(mockConnection);
+    accountDao = new AccountDao(dataAccessor);
+  }
+
+  /**
+   * Utility to get a {@link MySqlDataAccessor}.
+   * @param mockConnection the connection to use.
+   * @return the {@link MySqlDataAccessor}.
+   * @throws SQLException
+   */
+  static MySqlDataAccessor getDataAccessor(Connection mockConnection) throws SQLException {
+    Driver mockDriver = mock(Driver.class);
+    when(mockDriver.connect(anyString(), any(Properties.class))).thenReturn(mockConnection);
+    //lenient().when(mockConnection.isValid(anyInt())).thenReturn(true);
+    Properties properties = new Properties();
+    properties.setProperty(MySqlConfig.MYSQL_URL, "jdbc:mysql://localhost/AccountMetadata");
+    properties.setProperty(MySqlConfig.MYSQL_USER, "ambry");
+    properties.setProperty(MySqlConfig.MYSQL_PASSWORD, "ambry");
+    MySqlConfig config = new MySqlConfig(new VerifiableProperties(properties));
+    MySqlDataAccessor dataAccessor = new MySqlDataAccessor(config, mockDriver);
+    when(dataAccessor.getDatabaseConnection()).thenReturn(mockConnection);
+    return dataAccessor;
+  }
+
+  @Test
+  public void testAddAccount() throws Exception {
+    accountDao.addAccount(testAccount);
```

Comment:
Can we get the account to verify?

Reply:
Hard to do in a unit test since nothing is cached.  Integration test will do this.